lib/dash/dash_parser.js

```diff
@@ -992,6 +991,38 @@ shaka.dash.DashParser.prototype.parseAdaptationSet_ = function(context, elem) {
 
 
 /**
+ * Indicates whether an InbandEventStream element is present at the Adaption
+ * Set or Representation level.
+ *
+ * @param {!Element} elem The AdaptationSet element.
+ * @return {boolean} Whether the InbandEventStream element is present.
+ * @private
+ */
+shaka.dash.DashParser.prototype.inBandEventStreamIsPresent_ = function(elem) {
+  var XmlUtils = shaka.util.XmlUtils;
+
+  var adaptationEventStream = XmlUtils.findChild(elem, 'InbandEventStream');
+  if (adaptationEventStream != null) {
+    return true;
+  }
+
+  var representations = XmlUtils.findChildren(elem, 'Representation');
+  var representationEventStream = null;
+  if (representations.length > 0) {
+    for (var i = 0; i < representations.length; i++) {
+      representationEventStream =
+          XmlUtils.findChild(representations[i], 'InbandEventStream');
+
+      if (representationEventStream)
+        break;
```

Comment:
I could return true there and return false at the end of the function but I was trying to avoid so many return statements so I handled it with one return at the end of the function.